Source/Charts/Charts/ChartViewBase.swift

```diff
@@ -497,6 +497,22 @@ open class ChartViewBase: NSUIView, ChartDataProvider, AnimatorDelegate
     {
         var entry: ChartDataEntry?
         var h = highlight
+
+        if let xValue = h?.x
+        {
+            if let xValueIndex = xAxis.entries.index(of: xValue)
+            {
+                xAxis.selectedItemIndex = xValueIndex
+            }
+            else
+            {
+                xAxis.selectedItemIndex = NSInteger.min
+            }
+        }
+        else
+        {
+            xAxis.selectedItemIndex = NSInteger.min
```

Comment:
Why? If no value is selected or you can't get an entry for it, it should not select anything